Change location of PostgreSQL data

adamlazik1 opened the pull request:
The procedure in Administering Foreman, section 9.1 assumes
RHEL 7 disk layout. Added correct location of PostgreSQL data
for RHEL 8 and 9 while maintaining the original location for
RHEL 7.


* [x] I am familiar with the [contributing](https://github.com/theforeman/foreman-documentation/blob/master/CONTRIBUTING.md) guidelines.

Please cherry-pick 6e231a30c7525d7920c578289dc238626db9c335 into:

* [x] Foreman 3.3/Katello 4.5
* [x] Foreman 3.2/Katello 4.4
* [x] Foreman 3.1/Katello 4.3

Please cherry-pick fb40004da94195977b87a15f35cc76a840de108f into:

* [x] Foreman 3.3/Katello 4.5
* [ ] Foreman 3.2/Katello 4.4
* [ ] Foreman 3.1/Katello 4.3

ekohl commented on guides/common/modules/proc_estimating-the-size-of-a-backup.adoc:
```diff
@@ -32,11 +33,29 @@ A full backup requires space to store the following data:
 942M  total
 ----
++
+.For RHEL 8 and 9:
```
We don't have support for running on EL 9 yet

ekohl commented on guides/common/modules/proc_estimating-the-size-of-a-backup.adoc:
```diff
@@ -17,6 +17,7 @@ A full backup requires space to store the following data:
 .Procedure
 . Enter the `du` command to estimate the size of uncompressed directories containing {Project} database and configuration files:
 +
+.For RHEL 7:
```
We started a convention to use `{EL}` as a macro. For Satellite this expands to `{RHEL}` which expands to the full Red Hat Enterprise Linux. In upstream it expands to Enterprise Linux.

ekohl reviewed (commented):
I know it's a draft, but 2 small notes.

evgeni commented:
Ewouds comments are correct. Plus, we've been trying to mention EL8 first, as that's the primary deployment platform these days.

adamlazik1 commented:
@ekohl @evgeni Thank you both for your suggestions. 
Incorporated the suggested changes.

adamlazik1 commented:
Also I noticed that Satellite 6.12 supports only EL 8 while Foreman 3.3 still supports EL 7 and EL 8, so I guess that means I should make a separate PR to cherry-pick into 3.3?

evgeni commented:
Thanks! I am happy with the changes!

Any reason not to pick them to Foreman 3.3? Or was that an oversight?

adamlazik1 commented:
@evgeni I will make separate PR for 3.3, Because according to the milestones Foreman 3.3 supports both EL 7 and EL 8, while Satellite 6.12 only supports EL 8. I will need to put some `ifdef`s there. Please correct me if I am wrong. The alternative is to keep sections for EL 7 and EL 8 in Satellite 6.12 docs.

adamlazik1 commented:
Or maybe the right thing to do would be to update this PR by putting `ifdef`s with version numbers? But I am not sure whether that is a correct course of action since `ifdef`s with version numbers would break in the future. Let me know if you have any suggestions on how to resolve this.

evgeni commented:
The OS-support difference is correct. But I would just pick it now and add the `ifdefs` in a seperate PR?

adamlazik1 commented:
Sure thing. Updating PR then.

ekohl commented:
Indeed. It's fine now to write it as if we support both, pick that to all branches. Then later write it with ifdefs.

You can also write it in multiple commits:
* All changes that treat EL 7 & EL 8 as fully supported
* All changes that make it EL 7 & EL 8 upstream, EL 8 only in Satellite
* All changes that make it EL 8 only

Then we do selective cherry picks. IMHO it's fine to bundle that in a single PR to master and later raise cherry pick PRs for each individual branch.

adamlazik1 commented:
Removed unnecessary trailing slash from `/var/lib/pgsql/data/`. Added commit for Foreman 3.3 which supports both versions if `satellite` isn't defined. I made it so if any of `orcharhino`, `katello`, `foreman-deb` or `foreman-el` are defined, then sections for both EL 7 and EL 8 are present. Is this okay or should the section with EL 8-only display in some other than satellite build? Also made third commit in which the procedure only supports EL 8 as requested.

maximiliankolb commented on guides/common/modules/proc_estimating-the-size-of-a-backup.adoc:
```diff
@@ -42,10 +43,10 @@ The following table describes the compression ratio of all data items included i
 |80 - 85%
 |100 GB -> 20 GB
-
+    
```
Please remove this whitespace.

maximiliankolb commented on guides/common/modules/proc_estimating-the-size-of-a-backup.adoc:
```diff
@@ -32,6 +32,7 @@ A full backup requires space to store the following data:
 2M    /opt/puppetlabs
 942M  total
 ----
++
```
This should not be needed here. Please drop.

adamlazik1 commented:
@maximiliankolb Thank you for your feedback.
Incorporated the suggested changes

maximiliankolb commented:
Merged to master and cherry-picked: ✅ 

**Change location of PostgreSQL data**:
   8edd44b86..747c3929e  3.4 -> 3.4
   b07126902..d6cd0d768  3.3 -> 3.3
   adb34178e..a413c406b  3.2 -> 3.2
   1458d566d..d73b03f3c  3.1 -> 3.1

**Adjust the procedure for Foreman 3.3**:
   d6cd0d768..337dbdeec  3.3 -> 3.3

@adamlazik1 Please verify my cherry-picks.

adamlazik1 commented:
@maximiliankolb my apologies, I forgot that the 3.4 branch was added. Please also cherry-pick `Adjust the procedure for Foreman 3.3` into 3.4 and then cherry-pick `Remove support for EL 7 for future releases` on top of that. I will update the PR message too, just for clarification purposes.

maximiliankolb commented:
@adamlazik1 I have cherry-picked "Adjust the procedure for Foreman 3.3" and "Remove support for EL 7 for future releases" to 3.4:

   cfa2b304e..05bcbbcd9  3.4 -> 3.4

adamlazik1 commented:
@maximiliankolb Many thanks!
